docs/cip/CIP-8_Matrix_Packages.md

```diff
@@ -0,0 +1,209 @@
+TODO finish
+
+#### `data.digests`
+```yaml
+  digests:
+    data: "sha256:decafbad"
+```
+
+### `matrix.data.parquet`
+
+The `matrix.data.parquet` file contains the core dataset of the matrix package. Each row in this Parquet file closely mirrors Chroma database content. The following columns are expected:
+
+- `src`- **Original Document Source**: This field holds the original source data, which could be text, documents, or any other relevant information that forms the basis of the dataset. This maps directly to a Chroma document.
+- `embedding`- **Computed Embeddings**: Computed embeddings are generated from the original document source using the model specified in `matrix.package.yaml`. This maps directly to Chroma embeddings.
+- `meta`- **Metadata**: Metadata associated with each row is a set of key/value pairs that provide context or information about the data. This metadata could include timestamps, data source references, or any other relevant details. This maps directly to Chroma metadata.
+
+For now, the `matrix.data.parquet` file is required to physically exist adjacent to the `matrix.package.yaml` file. This is an initial implementation simplification and convenience. It is expected that a follow-up shortly after this CIP is implemented will add remote storage support, facilitated by the use of OCI registries.
+
+In this scenario, the actual `matrix.data.parquet` file would be stored remotely, and the matrix package would include references or pointers to the remote location rather than including the dataset itself. This approach would reduce package size and enhance efficiency in distributing and managing large datasets, making the data file optional in such cases.
+
+TODO address degrees of freedom with parquet - file metadata, multi-file splitting, etc.
+### Chroma Client & CLI Changes
+
+For matrix packages to be useful, the procedure for producing and consuming them must be trivial. This section describes changes to both the Chroma client and CLI in order to support matrix packages.
+
+TODO lollll so much to fill in - sdboyer can do some, but also others/jeffchuber?
+#### User story
+
+TODO describe basic user story/interaction loop, from initial data loading into chroma, massaging, then `chroma export-matrix` , then someone else doing a `chroma import-matrix`, then running their app
+
+#### Import and `chroma import-matrix`
+
+Idempotence is probably critical here
+
+TODO do this
+
+#### Export and `chroma export-matrix`
+
+TODO do this
+
+## Alternatives Considered
+
+This CIP is intended to set us on a good path toward tackling a large problem, not solve the problem outright. As such, some "alternatives" are actually paths we may plausibly explore in the future.
+
+### Alternate naming patterns
+
+Other naming patterns for matrix packages are possible. The form of name chosen has far-reaching impacts on how packages are distributed and consumed.
```

Comment:
Can we simply call them datasets, as is the industry standard?

Reply:
They also come with an index, and metadata about vectors, which feels enough to warrant some special naming.

Reply:
Agreed with @HammadB. i think datasets are a critically necessary part of a matrix package, but their value-add is specifically in the association between that dataset and some set of computed values. The essence of the package is an assertion by the package author that "this model, these configs, are the best way of representing this dataset."

Part of what's tricky there is that we're treating 'dataset' as opaque - is it the original raw dataset, or has it been processed, cleaned, chunked, etc.? That is _obviously_ part of the value in a package. And it clearly overlaps with the input pipelines topic in CIP 6. But i think we can keep it out of scope for now and return to it later.